.github/workflows/cppbuild.yml

```diff
@@ -0,0 +1,37 @@
+on: [push, pull_request]
+name: C++ Build
+jobs:
+  cmake-build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      max-parallel: 6
+      matrix:
+        os: [ubuntu-latest, windows-latest, macOS-latest]
+        compiler: [llvm, gnu]
+
+    steps:
+      - uses: actions/checkout@master
+        with:
+          submodules: true
+      - name: llvm build and test
+        if: matrix.compiler == 'llvm'
+        env:
+          CC: clang
+          CXX: clang++
+        run: |
+          mkdir ./cmake-build-debug
+          cd cmake-build-debug
+          cmake ../
+          cmake --build .
+        shell: bash
+      - name: gnu build and test
+        if: matrix.compiler == 'gnu'
+        env:
+          CC: gcc
+          CXX: g++
+        run: |
+          mkdir ./cmake-build-debug
+          cd cmake-build-debug
+          cmake ../
+          cmake --build .
+        shell: bash
```

Comment:
This entire build is awesome. Nicely done.

Reply:
My feeling after talking to Jamie is that it still needs some cleanup/tweaking because the build matrix includes windows but neither llvm nor gnu seem to be actually used when it does the windows build.   But since the build system is "working"(TM) I would wait and do that in a separate PR.

Reply:
But overall, yes it is a very cool achievement to have a combined python and c++ build complete with tests happening via github actions on all 3 platforms :)